### PR TITLE
Solver statistics

### PR DESCRIPTION
**Pull request by JoranHonig**
This pr implements the logic to measure query count and solving time.

Additionally two minor bugs were fixed

### CODE REVIEW

File: solidity_examples/BECToken.sol [hunk 1]
```diff
@@ -296,4 +295,4 @@ contract BecToken is PausableToken {
         //if ether is sent to this address, send it back.
         revert();
     }
-}
\ No newline at end of file
+}
```
Comment (nbanmp) [inline solidity_examples/BECToken.sol]: Good fix. :smile:

File: mythril/laser/smt/solver/solver_statistics.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+from time import time
+
+from mythril.support.support_utils import Singleton
+
+from typing import Callable
+
+
+def stat_smt_query(func: Callable):
```
Comment (nbanmp) [inline mythril/laser/smt/solver/solver_statistics.py]: Docstrings for functions / classes in this file? :smiley:

**Comment by nbanmp** [comment]
It might be nice to have even more fine grained reporting of solver time than the total. Perhaps we can have a list of solver times, and to append to that list each time the solver is called. WDYT?

**Comment by JoranHonig** [comment]
@nbanmp definately, depending on what we want we can extend the solver statistics & decorator to keep track of anything.

**Pull request merged**